modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/StatisticsEndpoint.java

```diff
@@ -0,0 +1,378 @@
+import org.opencastproject.security.api.SecurityService;
+import org.opencastproject.security.api.UnauthorizedException;
+import org.opencastproject.security.api.User;
+import org.opencastproject.statistics.api.DataResolution;
+import org.opencastproject.statistics.api.ResourceType;
+import org.opencastproject.statistics.api.StatisticsProvider;
+import org.opencastproject.statistics.api.StatisticsService;
+import org.opencastproject.statistics.api.TimeSeries;
+import org.opencastproject.statistics.api.TimeSeriesProvider;
+import org.opencastproject.util.RestUtil;
+import org.opencastproject.util.doc.rest.RestParameter;
+import org.opencastproject.util.doc.rest.RestQuery;
+import org.opencastproject.util.doc.rest.RestResponse;
+import org.opencastproject.util.doc.rest.RestService;
+
+import com.entwinemedia.fn.data.Opt;
+import com.google.gson.Gson;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+
+@Path("/")
+@RestService(name = "statistics", title = "statistics façade service",
+  abstractText = "Provides statistics",
+  notes = {"This service provides statistics."
+    + "<em>This service is for exclusive use by the module admin-ui. Its API might change "
+    + "anytime without prior notice. Any dependencies other than the admin UI will be strictly ignored. "
+    + "DO NOT use this for integration of third-party applications.<em>"})
+public class StatisticsEndpoint {
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory.getLogger(StatisticsEndpoint.class);
+  private static final String TIME_SERIES_PROVIDER_TYPE = "timeSeries";
+
+  private SecurityService securityService;
+  private IndexService indexService;
+  private AdminUISearchIndex searchIndex;
+  private StatisticsService statisticsService;
+
+  public void setSecurityService(SecurityService securityService) {
+    this.securityService = securityService;
+  }
+
+  public void setIndexService(IndexService indexService) {
+    this.indexService = indexService;
+  }
+
+  public void setSearchIndex(AdminUISearchIndex searchIndex) {
+    this.searchIndex = searchIndex;
+  }
+
+  public void setStatisticsService(StatisticsService statisticsService) {
+    this.statisticsService = statisticsService;
+  }
+
+  private static final class RawProviderQuery {
+    private String providerId;
```

Comment:
I'm confused - why do we need RawProviderQuery when we have ProviderQuery?

Reply:
I've added some Javadoc to clarify. We're using GSON to parse the JSON directly into a Java POJO (`RawProviderQuery`) and then, to actually work with it, we convert it to a proper query object (`ProviderQuery`).